areader/reader.go

```diff
@@ -1059,3 +1059,133 @@ func (ar *Reader) GetUpdateStorers() ([]handlers.UpdateStorer, error) {
 
 	return list, nil
 }
+
+func contains(val string, list []string) bool {
+	for _, listval := range list {
+		if val == listval {
+			return true
+		}
+	}
+	return false
+}
+
+func listUnionEmpty(a []string, b []string) bool {
+	for _, x := range a {
+		if contains(x, b) {
+			return false
+		}
+	}
+	return true
+}
+
+func insertValidElement(deps map[string]interface{}, key string, val interface{}) error {
+	// Type sentinel. Only string and []string allowed
+	switch val.(type) {
+	case string, []string:
+	default:
+		return fmt.Errorf("Invalid type: %T can not be inserted", val)
+	}
+	// If the key does not exist -> insert
```

Comment:
All these comments are left in here for readability of the PR, but will be removed before merging.